### PR TITLE
[prometheus-adapter] Add configuration item MetricsMaxAge

### LINKED ISSUE #2605: [prometheus-adapter] MetricsMaxAge should allow configuration
### Is your feature request related to a problem ?

MetricsMaxAge default value equal to MetricsRelistInterval, But MetricsMaxAge is no way to modify

### Describe the solution you'd like.

Set the default value in values, the deployment yaml refers to this value

### Describe alternatives you've considered.

NONE

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by xiaomudk**
#### What this PR does / why we need it
MetricsMaxAge default value equal to MetricsRelistInterval, But MetricsMaxAge is no way to modify

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes  #2605

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-adapter/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: prometheus-adapter
-version: 3.4.1
+version: 3.4.2
```
Comment (monotek) [inline charts/prometheus-adapter/Chart.yaml:3]: new feature should raise minor version

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-adapter/Chart.yaml (hunk 1 above)
Comment (xiaomudk) [reply charts/prometheus-adapter/Chart.yaml]: thanks, fix

File: charts/prometheus-adapter/templates/deployment.yaml [hunk 2]
```diff
@@ -55,6 +55,7 @@ spec:
         - --logtostderr=true
         - --prometheus-url={{ tpl .Values.prometheus.url . }}{{ if .Values.prometheus.port }}:{{ .Values.prometheus.port }}{{end}}{{ .Values.prometheus.path }}
         - --metrics-relist-interval={{ .Values.metricsRelistInterval }}
+        - --metrics-max-age={{ .Values.metricsMaxAge }}
```
Comment (steven-sheehy) [inline charts/prometheus-adapter/templates/deployment.yaml:58]: Why is this change necessary? Can't you just use `extraArguments` to pass `--metrics-max-age`?

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: charts/prometheus-adapter/templates/deployment.yaml (hunk 2 above)
Comment (xiaomudk) [reply charts/prometheus-adapter/templates/deployment.yaml:58]: Sorry, using extraArguments, is also possible
